Content.Server/GameObjects/Components/Weapon/Melee/MeleeWeaponComponent.cs

```diff
@@ -114,26 +113,82 @@ void IAttack.Attack(AttackEventArgs eventArgs)
 
                 if (entity.TryGetComponent(out IDamageableComponent damageComponent))
                 {
-                    damageComponent.ChangeDamage(DamageType.Blunt, Damage, false, Owner);
+                    damageComponent.ChangeDamage(DamageType, Damage, false, Owner);
                     hitEntities.Add(entity);
                 }
             }
 
-            if(!OnHitEntities(hitEntities, eventArgs)) return;
+            if(!OnHitEntities(hitEntities, eventArgs)) return true;
 
             if (Arc != null)
             {
-                var sys = _entitySystemManager.GetEntitySystem<MeleeWeaponSystem>();
-                sys.SendAnimation(Arc, angle, eventArgs.User, hitEntities);
+                var sys = EntitySystem.Get<MeleeWeaponSystem>();
+                sys.SendAnimation(Arc, angle, eventArgs.User, Owner, hitEntities);
             }
 
-            _lastAttackTime = IoCManager.Resolve<IGameTiming>().CurTime;
+            _lastAttackTime = curTime;
+            _cooldownEnd = _lastAttackTime + TimeSpan.FromSeconds(ArcCooldownTime);
 
             if (Owner.TryGetComponent(out ItemCooldownComponent cooldown))
             {
                 cooldown.CooldownStart = _lastAttackTime;
-                cooldown.CooldownEnd = _lastAttackTime + TimeSpan.FromSeconds(_cooldownTime);
+                cooldown.CooldownEnd = _cooldownEnd;
             }
+
+            return true;
+        }
+
+        bool IAttack.ClickAttack(AttackEventArgs eventArgs)
+        {
+            if (eventArgs.WideAttack) return false;
+
+            var curTime = IoCManager.Resolve<IGameTiming>().CurTime;
+
+            if(curTime < _cooldownEnd || !eventArgs.Target.IsValid())
+                return true;
```

Comment:
Same here.

Reply:
this needs to return true so it doesn't default to attacking with the fist